javascript/code-challenges/challenges-09.test.js

```diff
@@ -0,0 +1,315 @@
+'use strict';
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 1 - Review
+
+Write a function that finds the maximum value in an array
+using the 'reduce' method.
+
+E.g. [4,2,7,5,9,2] -> 9
+------------------------------------------------------------------------------------------------ */
+const maxInArray = (arr) => {
+  const maxFound = arr.reduce( (accum, val, idx) => {
+    return (accum > val ? accum : val);
+  });
+  return maxFound;
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 2
+
+Write a function named getCourseKeys that takes in the courseInfo object and returns an array containing the keys for the courseInfo object.
+
+For example: (['name', 'duration', 'topics', 'finalExam']).
+------------------------------------------------------------------------------------------------ */
+const courseInfo = { name: 'Code 301', duration: { dayTrack: '4 weeks', eveningTrack: '8 weeks'},
+  topics: ['SMACSS', 'APIs', 'NodeJS', 'SQL', 'jQuery', 'functional programming'],
+  finalExam: true
+};
+
+const getCourseKeys = (obj) => {
+  let temp = Object.keys(obj);
+  return temp;
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 3
+
+Write a function named checkValues that takes in an object and a value and returns true if the value is in the object.
+
+
+------------------------------------------------------------------------------------------------ */
+
+const checkValues = (obj, value) => {
+  let temp =''
+  Object.values(obj).forEach(value => {
+temp = value;
+  });
+  return (temp === value ? true : false);
```

Comment:
There's a helpful little thing called ".includes" you can use on objects. So you can just say
```js
const checkValues = (obj, value) => Object.value(obj).includes(value) ? true : false;
```